akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -448,31 +448,36 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
 
       // check by address without uid to make sure that node with same host:port is not allowed
       // to join until previous node with that host:port has been removed from the cluster
-      val alreadyMember = localMembers.exists(_.address == node.address)
-      val isUnreachable = !latestGossip.overview.reachability.isReachable(node)
+      localMembers.find(_.address == node.address) match {
+        case Some(m) if m.uniqueAddress == node ⇒
+          // node retried join attempt, probably due to lost Welcome message
+          logInfo("Existing member [{}] is joining again.", node)
+          if (node != selfUniqueAddress)
+            sender() ! Welcome(selfUniqueAddress, latestGossip)
+        case Some(m) ⇒
+          // node restarted, same host:port as existing member, but with different uid
+          // safe to down and later remove existing member
+          // new node will retry join
+          logInfo("New incarnation of existing member [{}] is trying to join. " +
+            "Existing will be removed from the cluster and then new member will be allowed to join.", node)
+          if (m.status != Down && m.status != Leaving && m.status != Exiting)
+            downing(m.address)
```

Comment:
how about the existing member is in the state of ```UP```?
how about a attack when someone signal a ```JOIN``` command?
cause akka2.4 will add support for binding on different address.I think we may think about this too.
different host : addr but same binding-host : binding-addr,this will cause the UP node down if that was an attack.

I think the downing the existing one should only progress when the existing one is ```UNREACHABLE```.